File: Typescript/AirGUI/DisplayObjectAnimator.ts
namespace AirGUI
{
    class DisplayObjectAnimatorProperty
    {
        public name:string;
        public path:Array<string>;
        public value:number;
        public startValue:number;
    }

    export class DisplayObjectAnimator
    {
        private t:number;
        private duration:number;

        private animations:Array<DisplayObjectAnimatorProperty>;

        constructor(private target:DisplayObject, data:any)
        {
            this.t = 0;
            this.duration = data.Duration;
            this.animations = data.animations;

            for(let a of this.animations)
            {
                a.path = a.name.split('.');
                let obj = this.target as any;
                for(let i = 0; i < a.path.length - 1; ++i)
                {
                    obj = obj[a.path[i]];
                }

                a.startValue = obj[a.path[a.path.length - 1]];
            }
        }

        public Update():boolean
        {
            this.t += window.app.deltaTime;
            this.t = Math.min(this.t, this.duration);

            for(let a of this.animations)
            {
                let v = Math.lerp(a.startValue, a.value, this.t / this.duration);

                let obj = this.target as any;
                for(let i = 0; i < a.path.length - 1; ++i)
                {
                    obj = obj[a.path[i]];
                }

                obj[a.path[a.path.length - 1]] = v;
            }

            return this.t >= this.duration;
        }
    }
}

File: C#/AirGUI/DisplayObjectAnimator.cs
using UnityEngine;
using System.Collections;
using Newtonsoft.Json;
using System.Collections.Generic;

namespace AirGUI
{
    public class DisplayObjectAnimator
    {
        public class Property
        {
            public string name;
            public float value;
        }

        [JsonProperty]
        private List<Property> animations;

        public float Duration
        {
            get; set;
        } = .25f;

        private DisplayObject target;

        public DisplayObjectAnimator(DisplayObject target)
        {
            this.target = target;

            animations = new List<Property>();
        }

        public DisplayObjectAnimator PositionX(float value)
        {
            animations.Add(new Property { name = "position.x", value = value });

            target.Position = new Vector2(value, target.Position.y);

            return this;
        }

        public DisplayObjectAnimator PositionY(float value)
        {
            animations.Add(new Property { name = "position.y", value = value });

            target.Position = new Vector2(target.Position.x, value);

            return this;
        }

        public DisplayObjectAnimator ScaleX(float value)
        {
            animations.Add(new Property { name = "scale.x", value = value });

            target.Scale = new Vector2(value, target.Position.y);

            return this;
        }

        public DisplayObjectAnimator ScaleY(float value)
        {
            animations.Add(new Property { name = "scale.y", value = value });

            target.Scale = new Vector2(target.Position.x, value);

            return this;
        }

        public DisplayObjectAnimator Alpha(float value)
        {
            animations.Add(new Property { name = "alpha", value = value });

            target.Alpha = value;

            return this;
        }
    } 
}
